src/test/groovy/com/stormpath/tck/authentication/CookieIT.groovy

```diff
@@ -91,54 +73,43 @@ class CookieIT extends AbstractIT {
         assertTrue(isCookieDeleted(response.detailedCookies.get("refresh_token")))
     }
 
-    /** Reject unauthorized text/html requests with 302 to login route
-     * 302 to login route should preserve original URL including query string
-     * @see <a href="https://github.com/stormpath/stormpath-framework-tck/issues/36">#36</a>
-     * @see <a href="https://github.com/stormpath/stormpath-framework-tck/issues/121">#121</a>
-     * @throws Exception
-     */
-    @Test(groups=["v100", "html"])
-    public void unauthorizedHtmlRequestIsForwardedToLogin() throws Exception {
-
-        // TODO: This test will probably have to change because we will be updating the spec.
-
-        given()
-            .accept(ContentType.HTML)
-        .when()
-            .get(MeRoute + "?foo=bar")
-        .then()
-            .statusCode(302)
-            .header("Location", is("/login?next=%2Fme%3Ffoo%3Dbar"))
-    }
-
     /** Cookie expires flag should be set to token TTL
      * @see <a href="https://github.com/stormpath/stormpath-framework-tck/issues/36">#38</a>
      * @throws Exception
      */
     @Test(groups=["v100", "json"])
     public void cookieExpirationMatchesTokenTtl() throws Exception {
         def account = new TestAccount()
+        saveCSRFAndCookies(FrameworkConstants.RegisterRoute)
+        account.setCSRF(csrf)
+        account.setCookies(cookies)
         account.registerOnServer()
         deleteOnClassTeardown(account.href)
 
-        def response = given()
+        saveCSRFAndCookies(FrameworkConstants.LoginRoute)
+
+        def requestSpecification = given()
             .accept(ContentType.JSON)
             .contentType(ContentType.JSON)
             .body([ "login": account.email, "password": account.password ])
-        .when()
-            .post(LoginRoute)
-        .then()
-            .statusCode(200)
-        .extract()
-            .response()
+
+        setCSRFAndCookies(requestSpecification, ContentType.JSON);
+
+        def response = requestSpecification
+            .when()
+                .post(LoginRoute)
+            .then()
+                .statusCode(200)
+            .extract()
+                .response()
 
         def accessTokenCookie = response.detailedCookies.get("access_token")
         def accessTokenTtl = JwtUtils.extractJwtClaim(accessTokenCookie.value, "exp") as long
-        assertEquals(accessTokenCookie.expiryDate, new Date(accessTokenTtl * 1000L))
+        assertTrue accessTokenTtl * 1000L - accessTokenCookie.expiryDate.time < 2000
```

Comment:
Let's discuss this drift directly. I am still unsure about why this is in the TCK.